presidio_evaluator/evaluation/evaluator.py

```diff
@@ -508,35 +584,97 @@ def generate_graph(title, tokens_df):
                         showgrid=False,  # Removes X-axis grid lines
                     ),
                     yaxis=dict(
-                        title=f"Tokens",
+                        title="Tokens",
                         linecolor="#BCCCDC",  # Sets color of X-axis line
                         showgrid=False,  # Removes X-axis grid lines
                     ),
+                    height=10 * len(tokens_df),
                 )
                 fig.update_layout(yaxis={"categoryorder": "total ascending"})
-                fig.show()
+                fig.show(save_as)
 
-            generate_graph(title="false-negatives", tokens_df=fns_tokens_df)
-            generate_graph(title="false-positives", tokens_df=fps_tokens_df)
+            fps_tokens_df = pd.concat(
+                [
+                    group_tokens(pd.read_csv(df_path), key="prediction")
+                    for df_path in fps_frames
+                ]
+            )
+            fns_tokens_df = pd.concat(
+                [
+                    group_tokens(pd.read_csv(df_path), key="annotation")
+                    for df_path in fns_frames
+                ]
+            )
+
+            generate_graph(
+                title="false-negatives", tokens_df=fns_tokens_df, key="annotation"
+            )
+            generate_graph(
+                title="false-positives", tokens_df=fps_tokens_df, key="prediction"
+            )
+
+        def plot_confusion_matrix(
+            self, entities: List[str],
+                confmatrix: List[List[int]],
+                save_as: Optional[str]=None
+        ) -> None:
+            # Create a DataFrame from the 2D list
+            confusion_matrix_df = pd.DataFrame(
+                confmatrix, index=entities, columns=entities
+            )
+
+            confusion_matrix_df.loc["Total"] = confusion_matrix_df.sum()
+
+            # Add a column for the totals
+            confusion_matrix_df["Total"] = confusion_matrix_df.sum(axis=1)
+
+            # Create the heatmap
+            fig = px.imshow(
+                confusion_matrix_df,
+                labels=dict(x="Predicted", y="Actual", color="Count"),
+                x=confusion_matrix_df.columns,
+                y=confusion_matrix_df.index,
+                color_continuous_scale="Blues",
+                title="Confusion Matrix",
+                text_auto=True,
+            )
+            fig.update_xaxes(tickangle=90, side="top", title_standoff=10)
+            fig.update_traces(textfont=dict(size=10))
+            fig.update_layout(width=800, height=800)
+
+            fig.show(save_as)
```

Comment:
Maybe adding error handling here to handle unexpected value of save_as paramter?

Reply:
`save_as` is handled by Plotly, so if someone passes a wrong value, plotly would return a message saying the value isn't correct